lib/gooddata/connection.rb

```diff
@@ -112,15 +117,14 @@ def sso_url(login, provider, opts = DEFAULT_SSO_OPTIONS)
     # @param [String] login Email address used for logging into gooddata
     # @param [String] provider Name of SSO provider
     # @return [GoodData::Rest::Client] Instance of REST client
-    def connect_sso(login, provider)
-      url = sso_url(login, provider)
-
-      params = {
-        :x_gdc_request => "#{GoodData::Rest::Connection.generate_string}:#{GoodData::Rest::Connection.generate_string}"
-      }
-
-      RestClient.get url, params do |response, _request, _result|
-        Rest::Client.connect_sso(:sst_token => URI.decode(response.cookies['GDCAuthSST']))
+    def connect_sso(login, provider, url, opts)
+      url, params = sso_url(login, provider, url)
+
+      RestClient::Request.execute(opts.merge({method: :post, url: url, payload: params})) do |response, _request, _result|
+        return Rest::Client.connect_sso(opts.merge({headers: {
+          x_gdc_authsst: response.cookies['GDCAuthSST'],
```

Comment:
Style/IndentHash: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.